forage-android/src/main/java/com/joinforage/forage/android/ui/ForageElement.kt

```diff
@@ -91,6 +91,14 @@ interface ForageElement<T : ElementState> {
      */
     fun setPosForageConfig(posForageConfig: PosForageConfig)
 
+    /**
+     * Explicitly request that the current input method's soft
+     * input be shown to the user, if needed. This only has an
+     * effect if the ForageElement is focussed, which can be
```

Comment:
I remember trying to put these into one method in the past and I think there was an issue where whatever tracks the focus of elements (some form of UI loop???) didn't update quickly enough to do both programmatically, but Devin could probably tell you better.

Reply:
Thanks for pointing out the type!

> Does the keyboard appearing imply that the input field has focus?
So you are right in that the keyboard will only be able to appear if the input is focused. However, there seems to be a significant precedent in Android that "showing the keyboard" does not imply you should focus the input; instead the convention appears that it should be a no-op if the keyboard is not focused. It's a bit bizarre in my opinion, too. I'd advocate we keep the `show keyboard ()` method to its current functionality because of that precedent though.